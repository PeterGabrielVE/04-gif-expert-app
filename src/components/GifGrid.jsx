
export const GifGrid = ({ category }) => {
    
    const getGifs = async() => {

        const url = `https://api.giphy.com/v1/stickers/search?api_key=7qXH8xWTO1wGqFPmD9JMMgdtAjKTpmKS&q=${ category }&limit=20`
        const resp = await fetch( url );
        const { data = [] } = await resp.json();


        const gifs = data.map( img => ({
            id: img.id,
            title: img.title,
            url: img.images.downsized_medium.url
        }));

        console.log(gifs);
    
    }
    
    getGifs();
    
    
    return (
      <>
          <h3>{ category }</h3>

      </>
    )
  }
  