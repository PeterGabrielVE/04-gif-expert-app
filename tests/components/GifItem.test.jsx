import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en  <GifItem />', () => { 
    
    const title = 'Athenas';
    const url = 'https://www.google.com.ve';

    test('debe hacer pruebas con el snapshot', () => { 

           const { container } = render( <GifItem title={ title } url={ url } /> );
           expect( container ).toMatchSnapshot();
     });
});