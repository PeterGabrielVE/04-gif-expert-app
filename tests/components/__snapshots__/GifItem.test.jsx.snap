// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifItem /> debe hacer pruebas con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Athenas"
      src="https://www.google.com.ve"
    />
    <p>
      Athenas
    </p>
  </div>
</div>
`;
